docs/decisions/0019-oep-58-atlas-translations-design.rst

```diff
@@ -0,0 +1,423 @@
+
+A new ``i18n_js_namespace`` property is needed for the :ref:`compile-js-command`
+to generate JavaScript translations in a centrally managed manner for all
+XBlocks as described in the :ref:`js-translations` section.
+
+The ``i18n_js_namespace`` property will eliminate the need to hardcode the
+namespace the `XBlock Makefile compile_translations rule`_.
+
+
+For example, the `Drag and Drop XBlock get_static_i18n_js_url`_ will need to
+be updated to support both the ``XBlockI18nService`` new
+``get_javascript_locale_path`` method and the namespace.
+
+.. code:: diff
+
+     class DragAndDropBlock(XBlock):
+
+   +   i18n_js_namespace = 'DragAndDropI18N'
+
+       @staticmethod
+       def _get_statici18n_js_url():
+           """
+           Returns the Javascript translation file for the currently selected language, if any found by
+           `pkg_resources`
+           """
+           lang_code = translation.get_language()
+           if not lang_code:
+               return None
+
+   +       # TODO: Make this the default once OEP-58 is implemented.
+   +       if hasattr(self.i18n_service, 'get_javascript_locale_path'):
+   +           atlas_locale_path = self.i18n_service.get_javascript_locale_path()
+   +           if atlas_locale_path:
+   +               return atlas_locale_path
+
+           text_js = 'public/js/translations/{lang_code}/text.js'
+           country_code = lang_code.split('-')[0]
+           for code in (translation.to_locale(lang_code), lang_code, country_code):
+               if pkg_resources.resource_exists(loader.module_name, text_js.format(lang_code=code)):
+                   return text_js.format(lang_code=code)
+           return None
+
+
+New ``compile_plugins_js_translations`` command
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This command will loop over XBlock modules that has the ``i18n_js_namespace``
+property and compile the JavaScript translations.
+
+For example if the Drag and Drop XBlock has the ``i18n_js_namespace``
+the ``compile_plugins_js_translations`` command will execute the following
+commands::
+
+  i18n_tool generate -v  # Generate the .mo files
+  python manage.py compilejsi18n --namespace DragAndDropI18N --output conf/plugins-locale/drag_and_drop_v2/js/
+
+
+Dismissed Proposals
+===================
+
+
+XBlocks and plugins have their own "atlas pull" command
+-------------------------------------------------------
+
+This dismissed proposal intends to have each XBlock and Plugin have their
+own ``make pull_translations`` and be responsible for managing pulling their
+own translations from the `openedx-translations repo`_.
+
+This proposal has been dismissed because it would require substantial work
+to get into the details for the ``lib/python3.8/site-packages/`` directory
+and ensure that the ``make pull_translations`` command won't corrupt the
+virtual environment.
+
+This is a non-trivial task and appears to add more complexity than necessary
+for not much added benefit.
```

Comment:
It'd be nice to elaborate a little bit here. I think the fact that the XBlocks/plugins won't be used outside the context of `edx-platform` makes it so having `edx-platform` manage the translations instead of the modules themselves helps explain the "not much added benefit" part.

Reply:
```suggestion
This is a non-trivial task and appears to add more complexity than necessary
due to the fact that XBlocks and plugins won't be used outside the context of ``edx-platform``.
```